### ISSUE TITLE
Different Style of Querying a Property (Two ways to query num. qubits)

### ISSUE THREAD

**Issue by ystade** [Munich-Quantum-Software-Stack/QDMI#16]
This issue is related to other issues and PRs (#2, #3, #4, #6). For clarity, I open this new issue.

At the moment there are in principle two ways to query the number of qubits, i.e., via

https://github.com/Munich-Quantum-Software-Stack/QDMI/blob/4d4d0dc7c8331f0b2dc43119f2da0a53d2a5c203/include/qdmi.h#L210

or

https://github.com/Munich-Quantum-Software-Stack/QDMI/blob/4d4d0dc7c8331f0b2dc43119f2da0a53d2a5c203/include/qdmi.h#L200

with 

```c
int num_qubits;
int QDMI_query_device_property_i(dev, NUM_QUBITS, &num_qubits);
```

However, for example the WMI backend only implements the first option. Besides the fact, that there should in all cases only be exact one way to query some property, I think those two methods also illustrate two different styles to query properties. I want the latter to be the center of this issue and discuss, the pros and cons of both approaches:

1. The first option introduces for every property its own function. This approach has the disadvantage that it is difficult to extend the interface since you have to define a new function for each property. On the other side, you have the advantage, that you can have custom parameters with respect to the property, i.e., a struct where you can represent a coupling map in a reasonable way.

2. For the second option you just need to add a new constant for new parameters and since those are just integers those new constants are not breaking interfaces from an older version, they will just not support the new properties. However, with this option I have my difficulties to think of a good solution to retrieve the coupling map for example since one would only have standard query function for the default datatype `int`, `float`, `double`, ...

To summarize, I think we should settle on one option and use it consistently and in particular prevent the cases where one property can be retrieved in two different ways as it is possible right now for the number of qubits.
